Packs/GenericSQL/Integrations/GenericSQL/GenericSQL.py

```diff
@@ -0,0 +1,199 @@
+from typing import Dict, Callable, Tuple, Any, List
+
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+''' IMPORTS '''
+import sqlalchemy
+import pymysql
+from sqlalchemy.sql import text
+
+# In order to use and convert from pymysql to MySQL this line is necessary
+pymysql.install_as_MySQLdb()
+
+
+class Client:
+    """
+    Client to use in the SQL databases integration. Overrides BaseClient
+    makes the connection to the DB server
+    """
+
+    def __init__(self, dialect: str, host: str, username: str, password: str, port: str,
+                 database: str, connect_parameters: str):
+        self.dialect = dialect
+        self.host = host
+        self.username = username
+        self.password = password
+        self.port = port
+        self.dbname = database
+        self.connect_parameters = connect_parameters
+        self.connection = self._create_engine_and_connect()
+
+    @staticmethod
+    def _convert_dialect_to_module(dialect: str) -> str:
```

Comment:
why this is a static method. but `generate_default_port_by_dialect` is not?
i believe this method does not belong in the class. @yuvalbenshalom WDYT?

Reply:
the way I saw it - "convert dialect" is getting a client parameter and providing what it needs in order to make a connection to the db (which is maid in the client) . 
"generate default" is making sure that in case the user forgot to put an input, the connection will use a default one, so it's more about getting the correct input. that's why I put one function in the client and the other outside, but logically both can be in the client too... what do you think?

Reply:
can leave with that stying here.